### DISCUSSION TITLE
SSH for containers?

### DISCUSSION THREAD

**Discussion by Xel-Naha** [bpg/terraform-provider-proxmox#784] [Q&A] [answered]
Hi again,

i've been trying to setup containers which require nesting and mountpoints. I understand this can't be done via the API, as it only root@pam has these privileges.

Reading the documentation it is my understanding that when configuring an ssh block in the provider that bgp could login using ssh.

"[SSH connection](https://registry.terraform.io/providers/bpg/proxmox/latest/docs#ssh-connection)

_The Proxmox provider can connect to a Proxmox node via SSH. This is used in the proxmox_virtual_environment_vm or proxmox_virtual_environment_file resource to execute commands on the node to perform actions that are not supported by Proxmox API. For example, to import VM disks, or to uploading certain type of resources, such as snippets.__"

do I understand correctly as well though that only VM's are supported and not CT/Containers as this is not (yet?) built in the the proxmox_virtual_environment_container bit? and if so, only ssh should be used?

thanks :)

**Comment by bpg-dev** [comment, marked as answer]
Hi @Xel-Naha!

You are correct that containers can only be managed via the Proxmox API. The SSH connection is used solely for VM and file resource management in very specific use cases. Essentially, it is a hack to address certain shortcomings in the PVE API (some of which may have already been addressed in PVE 8.x). I would rather minimize usage of SSH in the provider rather than expand it, for many reasons.

The feature you're asking for would involve the re-implementation of the majority of the PVE [`lxc` API](https://pve.proxmox.com/pve-docs/api-viewer/#/nodes/{node}/lxc/{vmid}) using the [`pct` CLI](https://pve.proxmox.com/pve-docs/pve-admin-guide.html#_strong_pct_strong_proxmox_container_toolkit) over SSH. This is a substantial undertaking, and at the moment, I don't have plans to pursue it.
